service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/CodeableConceptCdMapper.java

```diff
@@ -48,6 +49,12 @@ public String mapCodeableConceptToCd(CodeableConcept codeableConcept) {
         return TemplateUtils.fillTemplate(CODEABLE_CONCEPT_CD_TEMPLATE, builder.build());
     }
 
+    public String getDisplayFromCodeableConcept(CodeableConcept codeableConcept) {
+        return findMainCode(codeableConcept)
+            .map(cc -> findDisplayText(cc).orElse(StringUtils.EMPTY))
+            .orElse(StringUtils.EMPTY);
```

Comment:
Do you need this second orElse?

Reply:
as discussed, that orElse is in case findMainCode returns empty.